File: tests/typesafety/typescript/alias_repeated.ts
// expect: passes

export const fixedBackoff: (attempt: number, error: Error) => number = () => 30_000;

export const doublingBackoff: (attempt: number, error: Error) => number = (attempt) =>
  30_000 * 2 ** attempt;

export function runWithRetries(policy: (attempt: number, error: Error) => number): number {
  return policy(1, new Error("transient"));
}

File: tests/typesafety/python/alias_repeated.py
# expect: passes
# title: The same function type is written out three times
from collections.abc import Callable


def fixed_backoff(attempt: int, error: Exception) -> int:
    return 30_000


def doubling_backoff(attempt: int, error: Exception) -> int:
    return 30_000 * (1 << attempt)


def run_with_retries(policy: Callable[[int, Exception], int]) -> int:
    return policy(1, ValueError("transient"))
